Application/ControlMsrE2/ControlMsrE2.h

```diff
@@ -0,0 +1,133 @@
+ http://opensource.org/licenses/bsd-license.php
+ 
+ THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+ WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+ 
+ **/
+
+/*
+    Unless otherwise specified all data types are naturally aligned. Structures are aligned on boundaries equal to the largest internal datum of the structure and internal data are implicitly padded to achieve natural alignment.
+*/
+
+#include <Uefi.h>
+#include <PiDxe.h>
+#include <Library/PcdLib.h>
+#include <Library/UefiLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/OcMiscLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Register/Msr.h>
+#include <Protocol/MpService.h>
+
+#include <Library/MemoryAllocationLib.h>
+#include <Uefi/UefiInternalFormRepresentation.h>
+#include <Library/UefiHiiServicesLib.h>
+#include <Library/HiiLib.h>
+
+#include <Protocol/HiiFont.h>
+#include <Protocol/HiiImage.h>
+#include <Protocol/HiiImageEx.h>
+#include <Protocol/HiiImageDecoder.h>
+#include <Protocol/HiiString.h>
+#include <Protocol/HiiDatabase.h>
+#include <Protocol/HiiConfigRouting.h>
+#include <Protocol/HiiConfigAccess.h>
+#include <Protocol/HiiConfigKeyword.h>
+#include <Guid/HiiPlatformSetupFormset.h>
+
+#define DONT_STOP_AT 0xFFFF
+
+/*
+ Parameters to serch in a Form for
+ ifrHeaders which contain a Description
+ and to return the Varstore
+ */
+typedef struct {
+    EFI_STRING                   searchText;    // Text that must be part of the options description
+    EFI_HII_HANDLE               efiHandle;     // Handle of the list the Form is part of
+    EFI_HII_PACKAGE_LIST_HEADER *listHeader;    // Ptr to the contents of the handle
+    EFI_HII_PACKAGE_HEADER      *pkgHeader;     // Ptr to the current package in the list
+    EFI_IFR_OP_HEADER           *firstIfrHeader; // Ptr to first IfrHeader in Package
+    EFI_IFR_VARSTORE            *ifrVarStore;   // Ptr to ifrHeader of corresponding varstore in same list
+    EFI_IFR_ONE_OF              *ifrOneOf;      // Ptr to ifrHeader of BIOS Option
+    UINT16                       stopAt;        // Used to find a certain BIOS Option
+    UINT16                       count;         // Running number of suitable BIOS Options
+    
+} OneOfContext;
+
+/*
+ Callback - What to do, when a ifrHeader with a defined OP_CODE is found
+ ifrHeader   Current ifrHeader under scrutiny
+ stop        Ptr to Stop flag. If true no further search for opcodes. Can be NULL.
+ context     Ptr to Hanlder specific data
+ */
+typedef VOID OpCodeHandler (EFI_IFR_OP_HEADER* ifrHeader, UINT8* stop, VOID* context);
+
+/*
+ Commandline Arguments
+ */
+enum {
+    ARG_LOCK = 1,
+    ARG_UNLOCK = 2,
+    ARG_CHECK = 4,
+    ARG_INTERACTIVE = 8,
+    ARG_VERBOSE = 16
+};
+
+#define PADD(x,y) (void*)(((char*) x) + y)
+
+#define IS_VERBOSE() (mFlags & ARG_VERBOSE)
```

Comment:
For all non-boolean operations it is recommended to explicitly mark the operation result. I.e.  `((mFlags & ARG_VERBOSE) != 0)`.